lib/srv/regular/sshserver.go

```diff
@@ -337,9 +344,28 @@ func (s *Server) Shutdown(ctx context.Context) error {
 	// Stop heart beating immediately to prevent active connections
 	// from making the server appear alive and well.
 	if s.heartbeat != nil {
+		// TODO(espadolini): figure out if it's worth making HeartbeatI.Close
+		// block until all operations have ceased
```

Comment:
I'm not sure we want to address this in this PR, but even with context canceling I think that there could be a (very remote) chance that an `UpsertProxy` gets executed after the `DeleteProxy`.